idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java

```diff
@@ -126,9 +127,22 @@ public static String getLastVisitDate(MBPartner_BH patient) {
 
 	@Override
 	public Visit processEntity(String uuid, String docAction) throws Exception {
-		// We need to do something special for completing a sales order - do it
-		// asynchronously
-		if (StringUtil.isNullOrEmpty(docAction) || !docAction.equalsIgnoreCase(DocAction.ACTION_Complete)) {
+		int clientId = Env.getAD_Client_ID(Env.getCtx());
+		String clientIdsForSynchronousProcessingString =
+				MSysConfig_BH.getValue(MSysConfig_BH.CLIENT_IDS_FOR_SYNCHRONOUS_SALES_ORDER_PROCESSING, "");
+		List<Integer> clientIdsForSynchronousProcessing = new ArrayList<>();
+		try {
+			if (!StringUtil.isNullOrEmpty(clientIdsForSynchronousProcessingString)) {
+				clientIdsForSynchronousProcessing = Arrays.stream(clientIdsForSynchronousProcessingString.split(","))
+						.map(stringClientId -> Integer.parseInt(stringClientId.trim())).collect(Collectors.toList());
+			}
+		} catch (Exception exception) {
+			log.severe(exception.getMessage());
+		}
+		// We need to do something special for completing a sales order - do it asynchronously (except for the clients we
+		// want to do it synchronously for)
+		if (StringUtil.isNullOrEmpty(docAction) || (!docAction.equalsIgnoreCase(DocAction.ACTION_Complete) &&
+				!clientIdsForSynchronousProcessing.contains(clientId))) {
```

Comment:
We want this code block to execute if the client ID is specified for synchronous processing, so maybe:
```
if (StringUtil.isNullOrEmpty(docAction) || !docAction.equalsIgnoreCase(DocAction.ACTION_Complete) ||
				clientIdsForSynchronousProcessing.contains(clientId)))
```